src/pages/dashboard/panels/ImportBreakdownTabs/UnrecognizedTab.tsx

```diff
@@ -16,17 +15,26 @@ import Checkbox from '../../../../components/Input/Checkbox';
 import { markUnrecognizedFile } from '../../../../core/slices/mainpage';
 import UnrecognizedAvdumpedItem from './UnrecognizedAvdumpedItem';
 
+import { useGetFileUnrecognizedQuery, useLazyPostFileAVDumpQuery } from '../../../../core/rtkQuery/fileApi';
+
 function UnrecognizedTab() {
   const dispatch = useDispatch();
 
-  const avdumpKeyExists = useSelector((state: RootState) => !!state.localSettings.AniDb.AVDumpKey);
-  const avdumpList = useSelector((state: RootState) => state.mainpage.avdump);
-  const items = useSelector((state: RootState) => state.mainpage.unrecognizedFiles);
+  const [avdumpList, setAvdumpList] = useState([] as Array<string>);
+
+  const items = useGetFileUnrecognizedQuery({ pageSize: 0 });
   const itemsMarked = useSelector((state: RootState) => state.mainpage.unrecognizedMark);
+  const [avdumpTrigger, avdumpResult] = useLazyPostFileAVDumpQuery();
 
-  const runAvdump = (fileId: number) => dispatch(
-    { type: Events.UTILITIES_AVDUMP, payload: fileId },
-  );
+  const runAvdump = (fileId: number) => {
+    avdumpTrigger(fileId).then(() => {
+      if (avdumpResult.data?.Ed2k) {
+        const tempAvdumpList = avdumpList;
+        tempAvdumpList[fileId] = avdumpResult.data.Ed2k;
```

Comment:
also isn't this a mutation? 
I am not sure if this is Immer wrapped but if it isn't you need to do `const tempAvdumpList = [...avdumpList]` or Array.splice otherwise you are editing state directly.

Reply:
my bad 😞